src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,35 @@
+import { useParams } from 'react-router-dom'
+import { defaultURL } from '../utils'
+import { useState, useEffect } from 'react'
+
+export default function SingleArticle() {
+    const { id } = useParams()
+    const [ article, setArticle ] = useState([])
+    const [ loading, setLoading ] = useState(true)
+    const [ error, setError ] = useState(null)
+
+    useEffect(()=>{
+        defaultURL.get(`/articles/${id}`)
```

Comment:
I would have this behaviour happen in my utils file which handles api calls. Instead here i would expect to see a function call (eg fetchSingleArticle()). Then within that extracted function i would make the request to the defaultURL